crmsh/corosync.py

```diff
@@ -167,56 +184,101 @@ def __init__(self, ip, port=5403, algo="ffsplit", tie_breaker="lowest",
         self.mode = mode
+        self.qnetd_ip = None
 
     @property
     def qnetd_cacert_on_qnetd(self):
+        """
+        Return path of qnetd-cacert.crt on qnetd node
+        """
         return "{}/nssdb/{}".format(self.qnetd_path, self.qnetd_cacert_filename)
 
     @property
     def qnetd_cacert_on_local(self):
-        return "{}/{}/{}".format(self.qdevice_path, self.ip, self.qnetd_cacert_filename)
+        """
+        Return path of qnetd-cacert.crt on local node
+        """
+        return "{}/{}/{}".format(self.qdevice_path, self.qnetd_addr, self.qnetd_cacert_filename)
 
     @property
     def qnetd_cacert_on_cluster(self):
+        """
+        Return path of qnetd-cacert.crt on cluster node
+        """
         return "{}/{}/{}".format(self.qdevice_path, self.cluster_node, self.qnetd_cacert_filename)
 
     @property
     def qdevice_crq_on_qnetd(self):
+        """
+        Return path of qdevice-net-node.crq on qnetd node
+        """
         return "{}/nssdb/{}".format(self.qnetd_path, self.qdevice_crq_filename)
     
     @property
     def qdevice_crq_on_local(self):
+        """
+        Return path of qdevice-net-node.crq on local node
+        """
         return "{}/nssdb/{}".format(self.qdevice_path, self.qdevice_crq_filename)
 
     @property
     def qnetd_cluster_crt_on_qnetd(self):
+        """
+        Return path of cluster-cluster_name.crt on qnetd node
+        """
         return "{}/nssdb/cluster-{}.crt".format(self.qnetd_path, self.cluster_name)
 
     @property
     def qnetd_cluster_crt_on_local(self):
-        return "{}/{}/{}".format(self.qdevice_path, self.ip, os.path.basename(self.qnetd_cluster_crt_on_qnetd))
+        """
+        Return path of cluster-cluster_name.crt on local node
+        """
+        return "{}/{}/{}".format(self.qdevice_path, self.qnetd_addr, os.path.basename(self.qnetd_cluster_crt_on_qnetd))
 
     @property
     def qdevice_p12_on_local(self):
+        """
+        Return path of qdevice-net-node.p12 on local node
+        """
         return "{}/nssdb/{}".format(self.qdevice_path, self.qdevice_p12_filename)
 
     @property
     def qdevice_p12_on_cluster(self):
+        """
+        Return path of qdevice-net-node.p12 on cluster node
+        """
         return "{}/{}/{}".format(self.qdevice_path, self.cluster_node, self.qdevice_p12_filename)
 
-    def valid_attr(self, interfaces_inst):
-        if not bootstrap.package_is_installed("corosync-qdevice"):
+    def valid_attr(self):
+        """
+        Validate qdevice related options
+        """
+        if not utils.package_is_installed("corosync-qdevice"):
             raise ValueError("Package \"corosync-qdevice\" not installed on this node")
-        if self.ip == utils.this_node() or self.ip in interfaces_inst.ip_list:
+
+        try:
+            # socket.getaddrinfo works for both ipv4 and ipv6 address
+            res = socket.getaddrinfo(self.qnetd_addr, None)
+            self.qnetd_ip = res[0][-1][0]
```

Comment:
Could you put an example of this `res` object? To know what `[0][-1][0]` means

Reply:
I will put more comments here